python/paddle/v2/optimizer.py

```diff
@@ -27,6 +40,37 @@ def __impl__():
             __impl__)
         self.__opt_conf__ = swig_api.OptimizationConfig.createFromProto(
             self.__opt_conf_proto__)
+        self.__opt_conf_new__ = None
+
+    def set_remote_optimizer_config(self, **kwargs):
+        """
+            setup new golang pserver optimizer config proto
+        """
+        import py_paddle.swig_paddle as swig_api
+
+        def parse_list(values, message):
+            if isinstance(values[0], dict):
+                for v in values:
+                    cmd = message.add()
+                    parse_dict(v, cmd)
+            else:
+                message.extend(values)
+
+        def parse_dict(values, message):
+            for k, v in values.iteritems():
+                if isinstance(v, dict):
+                    parse_dict(v, getattr(message, k))
+                elif isinstance(v, list):
+                    parse_list(v, getattr(message, k))
+                else:
+                    setattr(message, k, v)
+
+        def dict_to_protobuf(value, message):
+            parse_dict(value, message)
+
+        # create new optimizer with proto config, add new optimizer here
+        self.__opt_conf_new__ = OptimizerConfig()
+        dict_to_protobuf(kwargs, self.__opt_conf_new__)
```

Comment:
Is the manually parsed optimizer configuration protobuf message human readable? 
how about replacing them with built-in python API `text_format`?  
https://stackoverflow.com/questions/33557965/print-human-friendly-protobuf-message
It can be printed in a pretty format, facilitate the future debug.